metadata-ingestion/src/datahub/utilities/sqlalchemy_query_combiner.py

```diff
@@ -213,7 +213,16 @@ def _handle_execute(
 
         # Figure out how many columns this query returns.
         # This also implicitly ensures that the typing is generally correct.
-        assert len(get_query_columns(query)) > 0
+        try:
+            assert len(get_query_columns(query)) > 0
+        except Exception as e:
```

Comment:
```suggestion
        except AttributeError as e:
```

I think this generally looks good, but I do want to keep the except clause fairly constrained so that we don't accidentally suppress other errors. `AttributeError` should catch any issues with query being a string or similar

If we tighten it to just attribute error, we can also remove the `logger.exception` call and only keep the `logger.debug` one.

Reply:
I like your suggestion, indeed, that is the error from our side and would still be catched by the code suggestion.